opentelemetry-api/src/opentelemetry/_metrics/__init__.py

```diff
@@ -129,17 +129,21 @@ def version(self):
     def schema_url(self):
         return self._schema_url
 
-    # FIXME check that the instrument name has not been used already
+    def _check_instrument_name(self, name):
```

Comment:
Since each Meter instance would get it's own set of `_instrument_names`, isn't there a case in which an instrument is created under `_ProxyMeter` and once `MeterProvider` is set, the same instrument name can be created under the real `Meter`?

Reply:
`ProxyX` should not create or do anything, but defer the action to someone else, hence the `Proxy` in its name. If there is no SDK set yet, `ProxyMeter` should defer the action to an API `NoOpMeter` who would have its own set of instrument names. Once the SDK is set, `ProxyMeter` should defer the action to the SDK `Meter` who would have its own set of instrument names.

This is what the [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#instrument) says:

> Meter implementations MUST return an error when multiple Instruments are registered under the same Meter instance using the same name.

If a meter is created with a certain name under the API `NoOpMeter` instance and then another instrument is created with the same name under the SDK `Meter` instance no exception will be raised and that is compliant with the spec because the instruments were registered under different Meter instances using the same name.

Reply:
Yes, that is fine with me :). Just from a user standpoint, it could be confusing because on the surface it looks like they are using a single instance of `Meter` but underneath there are two instances.

Reply:
@lzchen nice :sunglasses: are you ok with this conversation being resolved? if so, please do so and we can merge :+1: